File: missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
  
        int N = nums.size();
        int Totalsum = (N*(N+1))/2;
        int Currentsum =0;

        for(int i =0;i<N;i++){
           Currentsum+=nums[i];
        }

        return Totalsum-Currentsum;
    }
};

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int size = nums.length;
        HashSet<Integer> set = new HashSet<>();

        // Put all elements into the set
        for (int num : nums) {
            set.add(num);
        }

        // Check which number from 0 to n is missing
        for (int i = 0; i <= size; i++) {
            if (!set.contains(i)) {
                return i;
            }
        }
        return -1;
    }
}
